packaging/specifiers.py

```diff
@@ -965,7 +965,7 @@ def filter(
         if self._specs:
             for spec in self._specs:
                 iterable = spec.filter(iterable, prereleases=bool(prereleases))
-            return iterable
+            return iter(iterable)
```

Comment:
Could also do:
```suggestion
            yield from iterable
```

Works in the other cases as well.